eZ/Publish/Core/Persistence/Legacy/Content/Gateway/DoctrineDatabase.php

```diff
@@ -346,249 +267,246 @@ public function updateContent($contentId, MetadataUpdateStruct $struct, VersionI
-        $qName->prepare()->execute();
-
-        // Now update ezcontentobject_attribute for current version
-        // Create update query that will be reused
-        /** @var $qAttr \eZ\Publish\Core\Persistence\Database\UpdateQuery */
-        $qAttr = $this->dbHandler->createUpdateQuery();
-        $qAttr
-            ->update($this->dbHandler->quoteTable('ezcontentobject_attribute'))
+        $query->execute();
+    }
+
+    private function updateContentFieldsAlwaysAvailableFlag(
+        int $contentId,
+        int $versionNo,
+        bool $alwaysAvailable,
+        int $languageMask,
+        int $initialLanguageId
+    ): void {
+        $query = $this->connection->createQueryBuilder();
+        $expr = $query->expr();
+        $query
+            ->update(self::CONTENT_FIELD_TABLE)
             ->where(
-                $qAttr->expr->lAnd(
-                    $qAttr->expr->eq(
-                        $this->dbHandler->quoteColumn('contentobject_id'),
-                        $qAttr->bindValue($contentId, null, \PDO::PARAM_INT)
-                    ),
-                    $qAttr->expr->eq(
-                        $this->dbHandler->quoteColumn('version'),
-                        $qAttr->bindValue(
-                            $contentInfoRow['current_version'],
-                            null,
-                            \PDO::PARAM_INT
-                        )
-                    )
+                $expr->eq(
+                    'contentobject_id',
+                    $query->createNamedParameter($contentId, ParameterType::INTEGER, ':contentId')
+                )
+            )
+            ->andWhere(
+                $expr->eq(
+                    'version',
+                    $query->createNamedParameter($versionNo, ParameterType::INTEGER, ':versionNo')
                 )
             );
 
         // If there is only a single language, update all fields and return
-        if (!$this->languageMaskGenerator->isLanguageMaskComposite($contentInfoRow['language_mask'])) {
-            $qAttr->set(
-                $this->dbHandler->quoteColumn('language_id'),
-                $alwaysAvailable ?
-                    $qAttr->expr->bitOr($this->dbHandler->quoteColumn('language_id'), 1) :
-                    $qAttr->expr->bitAnd($this->dbHandler->quoteColumn('language_id'), -2)
-            );
-            $qAttr->prepare()->execute();
+        if (!$this->languageMaskGenerator->isLanguageMaskComposite($languageMask)) {
+            $query
+                ->set(
+                    'language_id',
+                    $alwaysAvailable
+                        ? $this->databasePlatform->getBitOrComparisonExpression(
+                        'language_id',
+                        ':languageMaskOperand'
+                    )
+                        : $this->databasePlatform->getBitAndComparisonExpression(
+                        'language_id',
+                        ':languageMaskOperand'
+                    )
```

Comment:
```suggestion
                    $alwaysAvailable
                        ? $this->databasePlatform->getBitOrComparisonExpression(
                            'language_id',
                            ':languageMaskOperand'
                        )
                        : $this->databasePlatform->getBitAndComparisonExpression(
                            'language_id',
                            ':languageMaskOperand'
                        )
```
maybe extracting this to variable will improve readability

Reply:
There's an issue with reformatting of that - PhpStorm has different... idea (pun intended) which will be annoying when working on unrelated changes. Looks like this expression is repeated 3 times, so I need to extract it anyway.

Reply:
Refactored via 51d6370.